elastic/integrations
# Update Elastic Package Registry integration to use new indexer metrics (get operations)

## What does this PR do?



This PR updates the Elastic Package Registry integration to take into account the new metrics related to get operations (duration histograms) from https://github.com/elastic/package-registry/pull/853.


## Checklist


## Author's Checklist


- [x] Updated metrics are retrieved and shown in the dashboard
- [x] Update docker tag to be used in tests (_dev/deploy/docker/Dockerfile)

## How to test this PR locally



1. Go to `packages/elastic_package_registry` and build the package (`elastic-package build`).
2. Boot up the Elastic stack using [elastic-package](https://github.com/elastic/elastic-package) from `packages/elastic_package_registry` folder.
3. Boot up the EPR service by using the command: `elastic-package service up -v` inside the `packages/elastic_package_registry` folder
4. Get the container name to be used in configuration. It should be something similar to: `elastic-package-service_elastic_package_registry_1`
5. Add EPR integration in Kibana:
    - Set as hosts: `http://elastic-package-service_elastic_package_registry_1:9100`
    - Go to Discover and check in `metrics-*`. It can be filtered by:
        - `service.type: elastic_package_registry`
        - `data_stream.dataset: elastic_package_registry.metrics`
6. You can run queries against the EPR service:
    - Check the exposed container locally using `docker ps`
      ```bash
       $ docker ps |grep elastic-package-service_elastic_package_registry
      eacd95c89dc3   elastic-package-service_elastic_package_registry       "./package-registry"     About a minute ago   Up About a minute (healthy)   8080/tcp, 0.0.0.0:53910->8081/tcp, 0.0.0.0:53911->9110/tcp   elastic-package-service_elastic_package_registry_1
      ```
    - Run curl commands to that port, in this example the port is 53910:
      ```bash
      curl -s http://localhost:53910/
      curl -s http://localhost:53910/health
      curl -s http://localhost:53910/search?prerelease=true
      ... 
      ```
 

## Related issues


- Relates https://github.com/elastic/package-registry/issues/797


## Screenshots


![elastic_package_registry-metrics-overview](https://user-images.githubusercontent.com/5330827/181299682-12ff679e-9f7a-4cf3-8c80-50a367657ece.png)
